CHANGELOG.md

```diff
@@ -34,6 +34,9 @@ how a consumer would use the library (e.g. adding unit tests, updating documenta
     - `search` to search for audit-logs.
     - `send-to` to send audit-logs to server.
 
+- `code42 legal-hold show` option:
```

Comment:
We do have a new `devices list` command that is merged by not yet released.

Would it make more sense to have a command `code42 devices list --legal-hold-uid <UID>` to instead list? Basically a command that filters devices based on a legal-hold uid.

Just want to make sure all the options are considered, I don't really understand the use-case.

Reply:
Or a `code42 legal-hold list-devices [MATTER-ID]`.  Trying to find a way to afford  output format options (At least table and csv).

Reply:
I agree, if we can find a way to structure this to add a format option, that would be helpful to most customers. 

Currently, the `legal-hold` show command outputs data in the following formats:

* Matter info - table
* Users - columns
* (optional) Preservation policy - JSON


What do you think about removing users from the `show` command and creating `code42 legal-hold list-users [Matter ID]` with an `--include-devices` flag? I think it makes sense to keep users and devices closely tied together, rather than in separate commands.

Reply:
I like that! @timabrmsn What are you thoughts on this?  I believe you were the one who implemented the show command.

Reply:
I actually think it makes more sense to add legal hold status as an option to the `devices list` command, since that would be a useful option for people who are creating a report of all their devices to know which ones are on legal hold quickly when taking inventory of other things. It could just add a `legalHoldMatterId` column to the output. 

And keeping users under `legal-hold show` still makes sense to me, since it's a high level overview of the matter, of which membership is a key detail. Otherwise the `show` command just becomes a single row of the `list` command with the option to also show preservation policy. 

`devices list` already has `--include-usernames` which makes a separate call to `api/Users` and then merges the DataFrames on `userId`, so adding legal hold status would be easy enough to add using the same logic.

The storage total report is an interesting idea, but might be better suited to include in `devices list` report also, so it can be used more generically and not tied directly to only legal hold. I'm thinking something like `--include-storage-totals` on `devices list` that just adds a column per device for the sum of all that device's archives. Then if someone wanted to get a report broken down by matter Id, they could just filter by matter_id and sum the total in the csv report.

Reply:
Ok, `devices` it is. The device pull actually includes `userUid` without the username option, so it's even easier to merge in the legal hold data. 

There are two main things that we want this to accomplish: 

* Limit the device list to devices on a specific legal hold matter
`devices --matter-uid` - option to limit device search base on a specific legal hold matter, adds in columns such as legalHoldStatus, legalHoldMatterId, legalHoldMatterName 

* Add in legal hold membership columns for whatever device query you're running. 
 `devices -include-legal-hold-membership` - option to option to include legal hold membership on to the devices table, adds same columns as above but doesn't limit the population

This feels a little redundant, so any feedback is appreciated. 

---

As for `storage-totals`: is a sum of all device's archives appropriate? (this is how I wrote my first draft). However, the current legal hold script does some logic to determine which destination is the most recent and complete, and only uses that archive size as the total storage.

Reply:
I think that since filtering by matter-id outside of the CLI is simple enough (i.e. grep/Select-String or directly in excel), can we leave that to the end-user? I'd like to not have our option list expand too much. And I have some ideas for more generic column/row filtering logic that could work on any cmd that outputs a dataframe. Something like `--filter 'matter-id == 42' that would convert to a pandas expression under the hood. Although I haven't started trying to implement that so I'm not sure how feasible it is...

Also, because a device can be part of multiple legal holds, we need to think of how to display that. Can we just have a single column that has a list of matter-uids the device is part of, maybe separated by semi-colons or something?

Reply:
Yeah, for storage totals I think adding a `usedStorage` column that sums the total combined archive size for a device would work, maybe we could also add an `archiveCount` column that gets added with that option?

Trying to keep what features we add as generic and multi-use-case friendly as possible.